akeneo-design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {FC} from 'react';
+import styled from 'styled-components';
+import {PAGINATION_SEPARATOR, PaginationItem} from './PaginationItem';
+
+type PaginationProps = {
+  /**
+   * The current page number (starting at 1).
+   */
+  currentPage: number;
+
+  /**
+   * The total number of results.
+   */
+  totalItems: number;
+
+  /**
+   * The number of items per page. Default to 25.
+   */
+  itemsPerPage?: number;
+
+  /**
+   * Handler called when a pagination item is clicked.
+   */
+  onClick: (page: number) => void;
+};
+
+const MAX_PAGINATION_ITEMS_WITHOUT_SEPARATOR = 4;
+
+const Pagination: FC<PaginationProps> = ({currentPage, totalItems, itemsPerPage = 25, onClick}) => {
+  if (itemsPerPage <= 0) {
+    return <></>;
+  }
+
+  const numberOfPages = Math.ceil(totalItems / itemsPerPage);
+
+  if (currentPage > numberOfPages) {
+    throw new Error('');
+  }
+
+  if (numberOfPages <= 1) {
+    return <></>;
+  }
+
+  const pages = computePages(currentPage, numberOfPages);
+
+  return (
+    <PaginationContainer>
+      {pages.map((page: number | string, index: number) => {
```

Comment:
```suggestion
      {pages.map((page: PageContent, index: number) => {
```

As you use this type multiple times, you could maybe create a dedicated type:
```
Type PageContent = string | number
```